nomad/state/state_store.go

```diff
@@ -1486,6 +1491,32 @@ func (s *StateStore) nestedUpsertEval(txn *memdb.Txn, index uint64, eval *struct
 	return nil
 }
 
+// updateEvalModifyIndex is used to update the modify index of an evaluation that has been
+// through a scheduler pass. This is done as part of plan apply. It ensures that when a subsequent
+// scheduler workers process a re-queued evaluation it sees any partial updates from the plan apply.
+func (s *StateStore) updateEvalModifyIndex(txn *memdb.Txn, index uint64, evalID string) error {
+	// Lookup the evaluation
+	existing, err := txn.First("evals", "id", evalID)
+	if err != nil {
+		return fmt.Errorf("eval lookup failed: %v", err)
+	}
+	if existing == nil {
+		return fmt.Errorf("[ERR] state_store: unable to find eval id %q", evalID)
```

Comment:
```
err := fmt.Errorf("[ERR] state_store: unable to find eval id %q", evalID)
s.logger.Printf("[ERR] state_store: %v", err)
return err
```

Reply:
For this one, do you mind if the second line was just `s.logger.Printf(err)` so it doesn't double print [ERR] state_store?